### ISSUE TITLE
lzma-conduit leaks memory on standard conduit usage

### ISSUE THREAD

**Issue by da-x** [alphaHeavy/lzma-conduit#4]
I suspect a space leak in the implementation. Here are steps to reproduce.

I used a 700MB uncompressed file I created under /tmp using a simple bash command: "tar -cf /tmp/compress-me.bin  /usr/bin /usr/include". And I added a test to compress that file:

```haskell
    prop_compressBig :: Property
    prop_compressBig = monadicIO $ do
      run . runResourceT $ CB.sourceFile "/tmp/compress-me.bin" C.$$ 
                 compress (Just 2) C.=$= CB.sinkFile "/tmp/compress-me.bin.compressed"
```

It should be space-efficient due to the conduit use of the pipeline. However, the process takes up close to 300 MB in resident memory after running for a minute (and this number is increasing):

      PID USER	PR  NI    VIRT    RES    SHR S  %CPU %MEM     TIME+ COMMAND
    13203 dan	20   0  628156 291080   5256 S 100.2  3.7   0:51.08 lzma-test	

Regular compression using 'xz' util (e.g. "cat /tmp/compress-me.bin | xz -2 > /tmp/compress-me.bin.compressed" shows that the process utilizes the configured dictionary size for the compression but no more - it's about 16MB by default on my machine.

I don't think it's due to garbage collection - I've tried various options to make it more aggressive without success.

**Comment by NathanHowell** [comment]
Finally got around to looking into this. I've pushed a fix to master, any chance you can confirm it fixes it for you?

**Comment by da-x** [comment]
Yes, tested now - it fixes.

**Comment by NathanHowell** [comment]
This has been released as v1.1.3 - http://hackage.haskell.org/package/lzma-conduit-1.1.3
